tracer/src/Datadog.Trace/Configuration/ApmTracingConfig.cs

```diff
@@ -0,0 +1,184 @@
+
+#pragma warning disable SA1402 // File must contain single type
+namespace Datadog.Trace.Configuration
+{
+    /// <summary>
+    /// Represents an APM_TRACING configuration with targeting information
+    /// </summary>
+    internal class ApmTracingConfig
+    {
+        public ApmTracingConfig(string configId, LibConfig libConfig, ServiceTarget? serviceTarget, K8sTargetV2? clusterTarget)
+        {
+            ConfigId = configId;
+            LibConfig = libConfig;
+            ServiceTarget = serviceTarget;
+            ClusterTarget = clusterTarget;
+        }
+
+        public string ConfigId { get; }
+
+        public LibConfig LibConfig { get; }
+
+        public ServiceTarget? ServiceTarget { get; }
+
+        public K8sTargetV2? ClusterTarget { get; }
+
+        /// <summary>
+        /// Gets the priority of this configuration based on targeting specificity.
+        /// Higher values = higher priority.
+        /// </summary>
+        public int Priority
+        {
+            get
+            {
+                var hasService = !string.IsNullOrEmpty(ServiceTarget?.Service) && ServiceTarget?.Service != "*";
+                var hasEnv = !string.IsNullOrEmpty(ServiceTarget?.Env) && ServiceTarget?.Env != "*";
+                var hasCluster = ClusterTarget != null;
+
+                return new ConfigurationTarget(hasService, hasEnv, hasCluster) switch
+                {
+                    (true, true, _) => 5, // Service+env (highest priority)
+                    (true, false, _) => 4, // Service only
+                    (false, true, _) => 3, // Env only
+                    (false, false, true) => 2, // Cluster
+                    (false, false, false) => 1 // Org level
+                };
+            }
+        }
+
+        /// <summary>
+        /// Checks if this configuration matches the current service and environment
+        /// </summary>
+        public bool Matches(string serviceName, string environment)
+        {
+            if (ServiceTarget == null)
+            {
+                return true; // Org-level config matches everything
+            }
+
+            var serviceMatches = string.IsNullOrEmpty(ServiceTarget?.Service) ||
+                                 ServiceTarget?.Service == "*" ||
+                                 ServiceTarget?.Service == serviceName;
+
+            var envMatches = string.IsNullOrEmpty(ServiceTarget?.Env) ||
+                             ServiceTarget?.Env == "*" ||
+                             ServiceTarget?.Env == environment;
+
+            return serviceMatches && envMatches;
+        }
+
+        /// <summary>
+        /// Merges this configuration with another, giving priority to the higher priority config
+        /// </summary>
+        public ApmTracingConfig MergeWith(ApmTracingConfig other)
+        {
+            var higherPriority = this.Priority >= other.Priority ? this : other;
+            var lowerPriority = this.Priority >= other.Priority ? other : this;
```

Comment:
```suggestion
            var (higherPriority, lowerPriority) = this.Priority >= other.Priority ? (this, other) : (other, this);
```

Reply:
I like it too but does not work in .net461 (am I missing something?)